flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -283,6 +287,7 @@ private void commitDeltaTxn(NavigableMap<Long, WriteResult> pendingResults, Stri
         // merged one will lead to the incorrect delete semantic.
         WriteResult result = e.getValue();
         RowDelta rowDelta = table.newRowDelta()
+            .validateFromSnapshot(lastCommittedSnapshotId)
```

Comment:
What will be the value of `lastCommittedSnapshotId` if I restore the job from a savepoint? 

Shouldn't we do something similar for it as we do for `maxCommittedCheckpointId` but here many things to consider as if we run expire snapshot maintenance procedure before restoring the job from the savepoint(https://github.com/apache/iceberg/issues/2482#issuecomment-897017924) 

```
this.lastCommittedSnapshotId = getLastCommittedSnapshotId(table);
```

Reply:
As I listed in https://github.com/apache/iceberg/pull/2867, restart or restore from checkpoint/savepoint, the `lastCommittedSnapshotId` will just simply reset to null, and `rowDelta` will travel all snapshot history to ensure data files still exist and guarantee all snapshot history are valid. If we delete the expired snapshots by maintenance procedure before restoring the job, the validation will travel to the snapshot which is commited by expire snapshot maintenance procedure. I think that whould be ok.

This optimization just trying to speed up commit result in runtime, and I want to keep it simple, so I think we don't need to save the `lastCommittedSnapshotId` and restore it from checkpoint/savepoint. but I want to hear @openinx @jackye1995 @stevenzwu @kbendick opinions.

Reply:
@Reo-LEI Can you please help me understand how the #2867 help in solving this [validation error](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java#L435) 

I agree with you that we can handle the `lastCommittedSnapshotId` in a separate PR and get this reviewed because this will really speed up the commit time which increases with time. I have seen my flink job checkpoint time increases to 10-15mins from 100-200ms

Reply:
@ayush-san  I'm very sorry, this is my mistake, I want to linked to https://github.com/apache/iceberg/issues/3102#issuecomment-919073751, but not #2867 .

Reply:
Yes, but with your PR we can run the snapshot expire task with the flink job running since you are updating the lastCommittedSnapshotId. The only case now left is that when we start the flink job from a checkpoint, we will encounter the same problem.

Reply:
> but with your PR we can run the snapshot expire task with the flink job running since you are updating the lastCommittedSnapshotId.

I think the snapshot expire task don’t cause the validation error is becasue the `validationHistory` will just travel to the `lastCommittedSnapshot` and stop valid the removed snapshots which is older than the `lastCommittedSnapshot`. But once the `lastCommittedSnapshot` be removed(e.g. the snapshot expire task run multiple time between two checkpoints) , you will encounter the validation error again.

I think store the `lastCommittedSnapshotId` is not a correct way to resolve the validation error(#2482). Because we cann't guarantee the `lastCommittedSnapshot` what we store its snapshot id will alway exists when we restore the flink job. The `lastCommittedSnapshot` probably has been removed when we restore the flink job, and the validation error will raise again. @ayush-san